### PR TITLE
API 8653 better abstractions

### PR DESCRIPTION
**Pull request by bschofield-va**
- More descriptive RPC usage for coverage controller
- Polish on Coverage transformer/controller to be easier to read, take less code.

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageTransformer.java [hunk 1]
```diff
@@ -164,21 +142,15 @@ private Coverage toCoverage(LhsLighthouseRpcGatewayResponse.FilemanEntry entry)
...
     return Coverage.builder()
         .id(patientCoordinateStringFrom(patientIcn, rpcResults.getKey(), entry.ien()))
-        .extension(
-            extensions(
-                entry.fields().get(InsuranceType.PHARMACY_PERSON_CODE),
-                entry.fields().get(InsuranceType.STOP_POLICY_FROM_BILLING)))
+        .extension(extensions(entry))
         .status(Coverage.Status.active)
-        .subscriberId(subscriberId(entry.fields().get(InsuranceType.SUBSCRIBER_ID)))
+        .subscriberId(entry.external(InsuranceType.SUBSCRIBER_ID).orElse(null))
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageTransformer.java:147]: Should this use the internal value to keep uniform? Subscriber id should have the same internal and external values.

File: vista-fhir-query-mock-services/pom.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
   <packaging>jar</packaging>
   <properties>
     <mockserver.version>5.11.2</mockserver.version>
-    <charon.version>4.0.18</charon.version>
+    <charon.version>4.0.19-SNAPSHOT</charon.version>
```
Comment (jhulbert-va) [inline vista-fhir-query-mock-services/pom.xml:15]: Gotta RIP that `SNAPSHOT` before mergin'.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/RpcGatewayTransformers.java [hunk 3]
```diff
@@ -49,6 +49,7 @@ public static boolean yesNoToBoolean(String zeroOrOne) {
    * A illegal argument exception that indicates the value from Vista is unexpected, e.g. receiving
    * a non-numeric value in a field that is 'guaranteed' to be a number.
    */
+  @Deprecated
```
[the remark this answers is not in the document]
Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/RpcGatewayTransformers.java:52]: it doesn't really matter... I just want compiler warnings until I've removed it completely

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageTransformer.java (hunk 1 above)
Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageTransformer.java:147]: I didn't change logic here, just tidied up how it got there.

Comment (jhulbert-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/coverage/R4CoverageTransformer.java:147]: Right, I was just thinking out loud.

**Pull request merged**